test/hotspot/jtreg/serviceability/jvmti/vthread/ThreadListStackTracesTest/ThreadListStackTracesTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8295976
+ * @summary GetThreadListStackTraces returns wrong state for blocked VirtualThread
+ * @requires vm.continuations
+ * @modules java.base/java.lang:+open
+ * @run main/othervm/native -agentlib:ThreadListStackTracesTest ThreadListStackTracesTest
+ */
+
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.*;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+class TestTask implements Runnable {
+    volatile boolean threadReady = false;
+
+    static void log(String msg) { System.out.println(msg); }
+
+    static void sleep(long millis) {
+        try {
+            Thread.sleep(millis);
+        } catch (InterruptedException e) {
+            throw new RuntimeException("Interruption in TestTask.sleep: \n\t" + e);
+        }
+    }
+
+    public void ensureReady(Thread vt, Thread.State expState) {
+        // wait while the thread is not ready or thread state is unexpected
+        while (!threadReady || (vt.getState() != expState)) {
+            sleep(1);
+        }
+    }
+
+    public void run() {
+        log("TestTask.run()");
+    }
```

Comment:
I think this should be an abstract method.

Reply:
Thanks. Fixed now.